### ISSUE TITLE
Campaign Launch failure

### ISSUE THREAD

**Issue by RetnuhGub** [gophish/gophish#1175]
Before filing a new issue, please use the search bar at the top of the browser to search for similar issues. Also, please make sure you have read our [documentation](https://getgophish.com/documentation/), which covers how to use Gophish.

What version of Gophish are you using?: 
6.0 Windows 64x

Brief description of the issue: 
I can send test emails, both through the Sending Profiles and the Campaign, and they arrive just fine. Starting a campaign causes the terminal to run into a runtime error and close out.

What are you expecting to see happen? :
A campaign to start sending out emails. I'm using our own host that has an Exchange email server, attempted to use gmail's smtp as well, but had other errors that I was unable to figure out.

What are you seeing happen?
The cmd.exe terminal closes. If I boot it back up and log back into the admin page, I have about 2 minutes to delete the campaign before it attempts to send another email and errors out again.

If this question is related to email templates or landing pages not working as expected, please provide your template or landing page below:

```html
Insert code here
```

Please provide any terminal output that may be relevant below:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x68 pc=0x5b59e9]

goroutine 890 [running]:
net/url.(*URL).Query(0x0, 0x10)
        /usr/local/go/src/net/url/url.go:996 +0x29
github.com/gophish/gophish/models.(*MailLog).Generate(0xc0421e96d0, 0xc04278c620, 0x0, 0xc04278c620)
        /go/src/github.com/gophish/gophish/models/maillog.go:227 +0x3c4
github.com/gophish/gophish/mailer.sendMail(0xabb020, 0xc04220a3c0, 0xab5d40, 0xc04276c038, 0xc042dffc20, 0x1, 0x1)
        /go/src/github.com/gophish/gophish/mailer/mailer.go:156 +0x2aa
github.com/gophish/gophish/mailer.(*MailWorker).Start.func1(0xabb020, 0xc04220a3c0, 0xc042dffc20, 0x1, 0x1)
        /go/src/github.com/gophish/gophish/mailer/mailer.go:92 +0x156
created by github.com/gophish/gophish/mailer.(*MailWorker).Start
        /go/src/github.com/gophish/gophish/mailer/mailer.go:85 +0x88

```
Insert output here
```

Please provide as many steps as you can to reproduce the problem:
 * Step 1 - Set up the Sending Profiles using an internal host and credentials
 * Step 2 - Copied a URL to act as the landing page (Western Digital's login portal) and removed only the WD header in the HTML to better disguise the page as a generic login portal.
* Step 3 - Created generic email template and linked the landing page per the documentation
* Step 4 - Landing page is hosted on my local machine's IP address with correct port mapping (theoretically. Haven't been able to test it yet)

If I've missed any vital information, please let me know! This seems to be an incredible tool and free! Thanks a bunch.

**Comment by jordan-wright** [comment]
Hi @RetnuhGub,

What are you using for the URL field when building a campaign? You'll want to make sure that it points to your Gophish instance, and is the fully qualified URL in the `http://your_ip_or_hostname:port/path` format.

**Comment by RetnuhGub** [comment]
I (kind of) figured out a fix in the meanwhile. I downloaded the 0.5 build and redesigned the campaign using the same information, and it was able to launch successfully. I did run into a problem with Exchange '07 web-based client breaking the link in my emails due to missing the http://, but I was able to resolve that once I figured out my error. The 0.5v has been completely successful - and for that, I thank you! What incredible software.

For 0.6, are you talking about the reference to the URL in the config file or the campaign settings?

I had changed the URL in the config to my local machine and pointed to an available port, but I used the shorthand {{.URL}} for the link, and I believe I missed labeling it as http or https in the protocol (it might have been set to "other"). I can verify that once this campaign is finished. It was probably user error somewhere, but in the event I had found a bug, I wanted to bring it up.
